Add 2 ACVIP examples to import examples wizard

asazonova opened the pull request:
ACVIP project

Add examples to import examples wizard
- simple example with ALISA
- example with error annex clause and software platform in aadl

fixes #2586

asazonova commented:
awaiting model from video #2

lwrage reviewed (changes requested):
- Some AADL files are not formatted properly. Could you check all of them, please?
- Some files contain a block of empty lines, where one is enough.
- The AADL files should have a copyright header (ask Sholom or John for the text).
- Please reformat the markdown files with line length 80 characters or so.
- Markdown file for ALISA example: remove link to file in github (it may change), is the directory really ALISA-example or ACVIP-ALISA?
- In readme use present tense, e.g., _contains files_ instead of _will contain files_.
- AADL files contain commented out lines. Are needed for the tutorial? If yes, add a text comment to each of these that tells the user in which step to remove the comments. If not, remove the commented out lines.

asazonova commented:
> * Some AADL files are not formatted properly. Could you check all of them, please? - done
> * Some files contain a block of empty lines, where one is enough. - done
> * The AADL files should have a copyright header (ask Sholom or John for the text). - emailed John, waiting on copyright text
> * Please reformat the markdown files with line length 80 characters or so. - done
> * Markdown file for ALISA example: remove link to file in github (it may change), is the directory really ALISA-example or ACVIP-ALISA? - link removed. ACVIP-ALISA is a main directory, that has 2 subdirectories under it(one of those is ALISA-example). I added a note in markdown file as well to say that project has 2 directories: 1 with aadl files and 1 with ALISA requirements
> * In readme use present tense, e.g., _contains files_ instead of _will contain files_. - done
> * AADL files contain commented out lines. Are needed for the tutorial? If yes, add a text comment to each of these that tells the user in which step to remove the comments. If not, remove the commented out lines. - deleted extra lines. done

lwrage commented on examples/org.osate.examples/examples/ACVIP_ALISA/doc/markdown/README.md:
```diff
@@ -0,0 +1,73 @@
+Within the "ALISA_Example" folder, there are several types of files:
+
+* "SEI.org" contains information about the stakeholders of the project
+* "isolleteSystemCategories.cat" contains categories that can be assigned to a
```
typo: isolette

lwrage commented on examples/org.osate.examples/examples/ACVIP_ALISA/doc/markdown/README.md:
```diff
@@ -0,0 +1,73 @@
+  Requirements can have variables that are then used for value assertion. The 
+  file "SysVerify.verify" can help automate the verification process of the
+  model
+* "SysMethods.methodregistry" allows you to specify the method that is run by
```
change _the method that is_ to _additional methods that can be run_

asazonova replied:
done

lwrage reviewed (changes requested):
A couple more minor things.

asazonova commented on examples/org.osate.examples/examples/ACVIP_ALISA/doc/markdown/README.md:
```diff
@@ -0,0 +1,73 @@
+  requirement to help group requirements together
+* Files that end in ".goaldoc" allow you to specify goals for the project.
+  There is currently only a single goal within "SCSFamilyGoals.goaldoc" but
+  more will be added down the line. 
```
changed to "There is currently only a single goal within "SCSFamilyGoals.goaldoc"" since we no longer plan to work on further examples

asazonova commented:
I made requested changes. also emailed DM team to see if i can re-upload the files (so files attached to DM and what released matches exactly)

lwrage reviewed (changes requested):
Merged, started OSATE, and imported examples. I found some issues
1. in ACVIP:
   - ACVIP project isn't recognozed as an AADL project (.project file missing)
   - content of doc/html is not "gitignored" but contains ARP4761 readme html file
   - Nothing really works because of the missing .project file
   - syntax error in SimpleSensor.aadl
   - Are there supposed to be two diagrams for the same component? The only difference is in the layout.
2. ACVIP ALISA
  - This imports one project ACVIP-ALISA but it looks like it should import two projects, ALISA_Example and SCSProject, instead. Because of this nothing really works, e.g., opening the diagram.

Could you check this, please, and make sure the example project import works as it should?

asazonova commented:
did a full build and ran it in new workspace location, i can see the help text and it imports the projects as expected.

> ACVIP ALISA
> This imports one project ACVIP-ALISA but it looks like it should import two projects, ALISA_Example and SCSProject, instead. 
Changed plugin file to include project paths separately so each one is imported as a separate project. Also changed some text in help file to alert user that 2 projects will be imported (so there's no confusion about expecting a project with ACVIP-ALISA name)

asazonova commented:
the only alert it shows now is in ALISA_Example project in SysReq.reqspec file as not being able to find implementation of isoletteSystem::isoletteSystem. It does go away if SCS_Project is manually added as a referenced project. However, currently import feature does not have a mechanism to link referenced projects.
